enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,557 @@
+updating: etcd (deflated 64%)
+updating: kube-apiserver (deflated 72%)
+```
+
+## Design Details
+
+### Open Questions [optional]
+
+1. UX design during install process as well as during failure (log collection). The Installer will dump
+(potentially prettified) controller logs. We expect that this question will become easier to answer further
+into the development process.
+
+2. Whether to use downstream OpenShift-specific `kube-apiserver` and `etcd` dependencies and how to source them?
+
+3. When should the Installer declare infrastructure provisioning failed?
+
+### Test Plan
+
+As this is replacing existing functionality in the Installer, we can rely on existing
+testing infrastructure.
+
+### Graduation Criteria
+
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+As this enhancement only concerns the Installation process and affects only the underlying cluster
+infrastructure, this change should not affect existing cluster upgrades.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+During a failed install, the controller logs will contain useful information. The status of the CAPI manifests
+may also contain useful information, in which case it would be important to display that to users and collect
+for bugs and support cases. There is an open question about the best way to handle this UX, and we expect the answer to become more clear during development.
+
+As the infrastructure will be reconciled by a controller, it will be possible to resolve issues during an ongoing
+installation, although this would not necessarily be a feature we would call attention to for documented use cases.
+
+Finally, the Installer will need to be able to identify when infrastructure provisioning has failed during an installation.
+Initially this will be achieved through a timeout. There is an open question about whether this can be done in a more sophisticated manner.
+
+#### Support Procedures
+
+When infrastrucutre provisioning has failed, users will need to collect the log bundle (see previous section
+regarding the open question for the log bundle).
+
+As the providers will be running in a control loop, it would be possible to resolve certain issues
+(e.g. fix missing permissions or delete resources taking up quota) during an installation, but this would
+not be a documented procedure. Furthermore, it would be possible to make the installs re-entrant, but
+it would not be a goal for simplicity's sake.
```

Comment:
Lets add that as a non-goal above. This was a question I had while reading until I got to this point.